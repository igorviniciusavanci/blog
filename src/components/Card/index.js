import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import styles from './style.module.css';

export default function Card({ title, content, publicationDate, author }) {
  function formatDate(timetamp) {
    return moment(timetamp).format('MMM DD, YYYY');
  }

  return (
    <li data-testid="card" className={styles.cardsItem}>
      <div className={styles.card}>
        <div className={styles.card_content}>
          <h1 className={styles.card_title}>{title}</h1>
          <p className={styles.card_text}>{content}</p>
          <p className={styles.author}>{author}</p>
          <p className={styles.date}>{formatDate(publicationDate)}</p>
          <button className={styles.btn}>Read More</button>
        </div>
      </div>
    </li>
  );
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  publicationDate: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
};
