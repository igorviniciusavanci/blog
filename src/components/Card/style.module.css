.btn {
  color: #0d0f1f;
  padding: 0.8rem;
  font-size: 14px;
  text-transform: uppercase;
  border-radius: 4px;
  font-weight: 400;
  display: block;
  width: 100%;
  cursor: pointer;
  border: 1px solid rgba(0, 0, 0, 0.2);
  background: transparent;
}

.btn:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

.author {
  font-size: 11px;
  font-weight: 700;
}
.date{
  font-size: 12px;
}

.cardsItem {
  display: flex;
  margin-top: 1rem;
  margin-right: 1rem;
}

.card {
  border-radius: 0.25rem;
  box-shadow: 0 20px 40px -14px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  overflow: hidden;
}

.card_content {
  padding: 1rem;
  background: #ffffff;
}

.card_title {
  color: #292929;
  font-size: 1.1rem;
  font-weight: 700;
  letter-spacing: 1px;
  text-transform: capitalize;
  margin: 0px;
}

.card_text {
  color: #888782;
  font-size: 0.875rem;
  line-height: 1.5;
  margin-bottom: 1.25rem;
  font-weight: 400;
}
