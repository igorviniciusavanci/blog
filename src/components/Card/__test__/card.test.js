import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import moment from 'moment';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Card from '..';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Card
      key="1"
      title="test"
      content="tesssssst"
      author="Igor"
      publicationDate={1492004832000}
    />,
    div
  );
});

it('render card correctly', () => {
  const { getByTestId } = render(
    <Card
      key="1"
      title="test"
      content="tesssssst"
      author="Igor"
      publicationDate={1492004832000}
    />
  );
  expect(getByTestId('card')).toHaveTextContent('test');
  expect(getByTestId('card')).toHaveTextContent('tesssssst');
  expect(getByTestId('card')).toHaveTextContent(
    moment(1492004832000).format('MMM DD, YYYY')
  );
});

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Card
        key="1"
        title="test"
        content="tesssssst"
        author="Igor"
        publicationDate={1492004832000}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
