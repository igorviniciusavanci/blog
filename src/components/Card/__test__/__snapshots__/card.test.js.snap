// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  className="cardsItem"
  data-testid="card"
>
  <div
    className="card"
  >
    <div
      className="card_content"
    >
      <h1
        className="card_title"
      >
        test
      </h1>
      <p
        className="card_text"
      >
        tesssssst
      </p>
      <p
        className="author"
      >
        Igor
      </p>
      <p
        className="date"
      >
        Apr 12, 2017
      </p>
      <button
        className="btn"
      >
        Read More
      </button>
    </div>
  </div>
</li>
`;
